### ISSUE TITLE
Adding a BlockCode node to any scene (2D or 3D) locks up the editor.

### ISSUE THREAD

**Issue by Naros** [endlessm/godot-block-coding#101]
### What Happened

Just adding a new `BlockCode` node locks up Godot.

### How to Reproduce

1. Create a simple 2D or 3D scene with a single root node.
2. Click root node, add a new node, selecting `BlockCode`.
3. Nothing happens, editor freezes

### Operating System

Windows, Ubuntu, Fedora

**Comment by dsd** [comment]
That's strange. Thanks for reporting. Which Godot and block coding plug versions are you running?

**Comment by Naros** [comment]
Godot 4.2.2-stable and Godot 4.3.beta2, and BlockCode 0.3.1 according to the `plugin.cfg`.

**Comment by manuq** [comment]
> Godot 4.2.2 stable and Godot 4.3 beta2, using BlockCode 0.3.1 according to the `plugin.cfg`.

The BlockCode plugin doesn't work with Godot 4.3 yet. If you opened the project in such version the project got converted, and that's probably why it doesn't work in 4.2 too. Please stick to 4.2 projects, hopefully we'll make it work with the 4.3 soon!

**Comment by Naros** [comment]
Even in a fresh Godot 4.2.2-stable install:
```
Godot Engine v4.2.2.stable.official (c) 2007-present Juan Linietsky, Ariel Manzur & Godot Contributors.
  modules/gltf/register_types.cpp:63 - Blend file import is enabled in the project settings, but no Blender path is configured in the editor settings. Blend files will not be imported.
  res://addons/block_code/ui/main_panel.gd:5 - Parse Error: Could not find type "EditorInterfaceAccess" in the current scope.
  res://addons/block_code/ui/main_panel.gd:7 - Parse Error: Could not find type "Picker" in the current scope.
  res://addons/block_code/ui/main_panel.gd:8 - Parse Error: Could not find type "BlockCanvas" in the current scope.
  res://addons/block_code/ui/main_panel.gd:9 - Parse Error: Could not find type "DragManager" in the current scope.
  res://addons/block_code/ui/main_panel.gd:10 - Parse Error: Could not find type "TitleBar" in the current scope.
  res://addons/block_code/ui/main_panel.gd:14 - Parse Error: Could not find type "BlockCode" in the current scope.
  res://addons/block_code/ui/main_panel.gd:52 - Parse Error: Could not find type "BlockCode" in the current scope.
  res://addons/block_code/ui/main_panel.gd:27 - Parse Error: Identifier "EditorInterfaceAccess" not declared in the current scope.
  res://addons/block_code/ui/main_panel.gd:38 - Parse Error: Could not find type "BlockScriptData" in the current scope.
  res://addons/block_code/ui/main_panel.gd:53 - Parse Error: Could not find type "BlockScriptData" in the current scope.
  res://addons/block_code/ui/main_panel.gd:67 - Parse Error: Could not find type "BlockScriptData" in the current scope.
  res://addons/block_code/ui/main_panel.gd:73 - Parse Error: Cannot infer the type of "block_trees" variable because the value doesn't have a set type.
  res://addons/block_code/ui/main_panel.gd:99 - Parse Error: Could not find type "BlockScriptData" in the current scope.
  modules/gdscript/gdscript.cpp:2726 - Failed to load script "res://addons/block_code/ui/main_panel.gd" with error "Parse error". (User)
  res://addons/block_code/ui/picker/categories/block_category_display.gd:7 - Parse Error: Could not find type "BlockCategory" in the current scope.
  res://addons/block_code/ui/picker/categories/block_category_display.gd:39 - Parse Error: Identifier "BlockCategory" not declared in the current scope.
  res://addons/block_code/ui/picker/categories/block_category_display.gd:44 - Parse Error: Could not find type "Block" in the current scope.
  res://addons/block_code/ui/picker/categories/block_category_display.gd:44 - Parse Error: Could not find type "Block" in the current scope.
  modules/gdscript/gdscript.cpp:2726 - Failed to load script "res://addons/block_code/ui/picker/categories/block_category_display.gd" with error "Parse error". (User)
  res://addons/block_code/ui/picker/picker.gd:5 - Parse Error: Could not find type "Block" in the current scope.
  res://addons/block_code/ui/picker/picker.gd:10 - Parse Error: Could not find type "BlockScriptData" in the current scope.
  res://addons/block_code/ui/picker/picker.gd:40 - Parse Error: Could not find type "Block" in the current scope.
  res://addons/block_code/ui/picker/picker.gd:40 - Parse Error: Could not find type "BlockCategory" in the current scope.
  res://addons/block_code/ui/picker/picker.gd:59 - Parse Error: Could not find type "Block" in the current scope.
  res://addons/block_code/ui/picker/picker.gd:15 - Parse Error: Could not find type "Block" in the current scope.
  res://addons/block_code/ui/picker/picker.gd:16 - Parse Error: Could not find type "BlockCategory" in the current scope.
  res://addons/block_code/ui/picker/picker.gd:30 - Parse Error: Identifier "CategoryFactory" not declared in the current scope.
  res://addons/block_code/ui/picker/picker.gd:43 - Parse Error: Identifier "CategoryFactory" not declared in the current scope.
  res://addons/block_code/ui/picker/picker.gd:43 - Parse Error: Cannot infer the type of "blocks" variable because the value doesn't have a set type.
  res://addons/block_code/ui/picker/picker.gd:44 - Parse Error: Identifier "CategoryFactory" not declared in the current scope.
  res://addons/block_code/ui/picker/picker.gd:44 - Parse Error: Cannot infer the type of "block_categories" variable because the value doesn't have a set type.
  res://addons/block_code/ui/picker/picker.gd:47 - Parse Error: Could not find type "BlockCategory" in the current scope.
  res://addons/block_code/ui/picker/picker.gd:47 - Parse Error: Could not find type "BlockCategory" in the current scope.
  res://addons/block_code/ui/picker/picker.gd:55 - Parse Error: Could not find type "Block" in the current scope.
  res://addons/block_code/ui/picker/picker.gd:55 - Parse Error: Could not find type "Block" in the current scope.
  modules/gdscript/gdscript.cpp:2726 - Failed to load script "res://addons/block_code/ui/picker/picker.gd" with error "Parse error". (User)
  res://addons/block_code/ui/title_bar/title_bar.gd:24 - Parse Error: Could not find type "BlockScriptData" in the current scope.
  res://addons/block_code/ui/title_bar/title_bar.gd:53 - Parse Error: Could not find type "BlockScriptData" in the current scope.
  res://addons/block_code/ui/title_bar/title_bar.gd:19 - Parse Error: Could not find type "BlockCode" in the current scope.
  res://addons/block_code/ui/title_bar/title_bar.gd:62 - Parse Error: Could not find type "BlockCode" in the current scope.
  modules/gdscript/gdscript.cpp:2726 - Failed to load script "res://addons/block_code/ui/title_bar/title_bar.gd" with error "Parse error". (User)
  res://addons/block_code/ui/block_canvas/block_canvas.gd:15 - Parse Error: Could not find type "Block" in the current scope.
  res://addons/block_code/ui/block_canvas/block_canvas.gd:32 - Parse Error: Could not find type "Block" in the current scope.
  res://addons/block_code/ui/block_canvas/block_canvas.gd:39 - Parse Error: Could not find type "Block" in the current scope.
  res://addons/block_code/ui/block_canvas/block_canvas.gd:39 - Parse Error: Could not find type "Block" in the current scope.
  res://addons/block_code/ui/block_canvas/block_canvas.gd:50 - Parse Error: Could not find type "BlockScriptData" in the current scope.
  res://addons/block_code/ui/block_canvas/block_canvas.gd:79 - Parse Error: Could not find type "SerializedBlockTreeNode" in the current scope.
  res://addons/block_code/ui/block_canvas/block_canvas.gd:94 - Parse Error: Could not find type "SerializedBlockTreeNodeArray" in the current scope.
  res://addons/block_code/ui/block_canvas/block_canvas.gd:102 - Parse Error: Could not find type "Block" in the current scope.
  res://addons/block_code/ui/block_canvas/block_canvas.gd:102 - Parse Error: Could not find type "SerializedBlockTreeNode" in the current scope.
  res://addons/block_code/ui/block_canvas/block_canvas.gd:81 - Parse Error: Could not find type "Block" in the current scope.
  res://addons/block_code/ui/block_canvas/block_canvas.gd:87 - Parse Error: Could not find type "Block" in the current scope.
  res://addons/block_code/ui/block_canvas/block_canvas.gd:87 - Parse Error: Could not find type "Block" in the current scope.
  res://addons/block_code/ui/block_canvas/block_canvas.gd:95 - Parse Error: Identifier "SerializedBlockTreeNodeArray" not declared in the current scope.
  res://addons/block_code/ui/block_canvas/block_canvas.gd:95 - Parse Error: Cannot infer the type of "block_trees" variable because the value doesn't have a set type.
  res://addons/block_code/ui/block_canvas/block_canvas.gd:103 - Parse Error: Identifier "SerializedBlockTreeNode" not declared in the current scope.
  res://addons/block_code/ui/block_canvas/block_canvas.gd:104 - Parse Error: Identifier "SerializedBlock" not declared in the current scope.
  res://addons/block_code/ui/block_canvas/block_canvas.gd:108 - Parse Error: Could not find type "Block" in the current scope.
  res://addons/block_code/ui/block_canvas/block_canvas.gd:130 - Parse Error: Could not find type "EntryBlock" in the current scope.
  res://addons/block_code/ui/block_canvas/block_canvas.gd:134 - Parse Error: Identifier "DragManager" not declared in the current scope.
  res://addons/block_code/ui/block_canvas/block_canvas.gd:134 - Parse Error: Cannot infer the type of "tree_scope" variable because the value doesn't have a set type.
  modules/gdscript/gdscript.cpp:2726 - Failed to load script "res://addons/block_code/ui/block_canvas/block_canvas.gd" with error "Parse error". (User)
  res://addons/block_code/ui/block_canvas/node_block_canvas/node_block_canvas.gd:3 - Parse Error: Could not find base class "BlockCanvas".
  modules/gdscript/gdscript.cpp:2726 - Failed to load script "res://addons/block_code/ui/block_canvas/node_block_canvas/node_block_canvas.gd" with error "Parse error". (User)
  res://addons/block_code/drag_manager/drag_manager.gd:165 - Parse Error: Could not find type "Picker" in the current scope.
  res://addons/block_code/drag_manager/drag_manager.gd:166 - Parse Error: Could not find type "BlockCanvas" in the current scope.
  res://addons/block_code/drag_manager/drag_manager.gd:198 - Parse Error: Could not find type "Block" in the current scope.
  res://addons/block_code/drag_manager/drag_manager.gd:198 - Parse Error: Could not find type "Block" in the current scope.
  res://addons/block_code/drag_manager/drag_manager.gd:223 - Parse Error: Could not find type "Block" in the current scope.
  res://addons/block_code/drag_manager/drag_manager.gd:223 - Parse Error: Could not find type "Block" in the current scope.
  res://addons/block_code/drag_manager/drag_manager.gd:227 - Parse Error: Could not find type "Block" in the current scope.
  res://addons/block_code/drag_manager/drag_manager.gd:251 - Parse Error: Could not find type "Block" in the current scope.
  res://addons/block_code/drag_manager/drag_manager.gd:267 - Parse Error: Could not find type "Block" in the current scope.
  res://addons/block_code/drag_manager/drag_manager.gd:267 - Parse Error: Could not find type "Block" in the current scope.
  res://addons/block_code/drag_manager/drag_manager.gd:18 - Parse Error: Could not find type "Block" in the current scope.
  res://addons/block_code/drag_manager/drag_manager.gd:20 - Parse Error: Could not find type "BlockCanvas" in the current scope.
  res://addons/block_code/drag_manager/drag_manager.gd:30 - Parse Error: Could not find type "SnapPoint" in the current scope.
  res://addons/block_code/drag_manager/drag_manager.gd:38 - Parse Error: Could not find type "Block" in the current scope.
  res://addons/block_code/drag_manager/drag_manager.gd:42 - Parse Error: Could not find type "Block" in the current scope.
  res://addons/block_code/drag_manager/drag_manager.gd:42 - Parse Error: Could not find type "BlockCanvas" in the current scope.
  res://addons/block_code/drag_manager/drag_manager.gd:52 - Parse Error: Could not find type "Block" in the current scope.
  res://addons/block_code/drag_manager/drag_manager.gd:133 - Parse Error: Could not find type "SnapPoint" in the current scope.
  res://addons/block_code/drag_manager/drag_manager.gd:133 - Parse Error: Could not find type "SnapPoint" in the current scope.
  res://addons/block_code/drag_manager/drag_manager.gd:136 - Parse Error: Could not find type "SnapPoint" in the current scope.
  res://addons/block_code/drag_manager/drag_manager.gd:208 - Parse Error: Could not find type "SnapPoint" in the current scope.
  res://addons/block_code/drag_manager/drag_manager.gd:228 - Parse Error: Could not find type "Block" in the current scope.
  res://addons/block_code/drag_manager/drag_manager.gd:256 - Parse Error: Could not find type "StatementBlock" in the current scope.
  res://addons/block_code/drag_manager/drag_manager.gd:257 - Parse Error: Could not find type "StatementBlock" in the current scope.
  res://addons/block_code/drag_manager/drag_manager.gd:257 - Parse Error: Cannot infer the type of "statement_block" variable because the value doesn't have a set type.
  res://addons/block_code/drag_manager/drag_manager.gd:259 - Parse Error: Could not find type "ParameterInput" in the current scope.
  res://addons/block_code/drag_manager/drag_manager.gd:260 - Parse Error: Cannot infer the type of "copy_block" variable because the value doesn't have a set type.
  res://addons/block_code/drag_manager/drag_manager.gd:264 - Parse Error: Could not find type "Block" in the current scope.
  res://addons/block_code/drag_manager/drag_manager.gd:268 - Parse Error: Could not find type "EntryBlock" in the current scope.
  res://addons/block_code/drag_manager/drag_manager.gd:275 - Parse Error: Could not find type "Block" in the current scope.
  res://addons/block_code/drag_manager/drag_manager.gd:86 - Parse Error: Could not find type "SnapPoint" in the current scope.
  res://addons/block_code/drag_manager/drag_manager.gd:86 - Parse Error: Could not find type "SnapPoint" in the current scope.
  res://addons/block_code/drag_manager/drag_manager.gd:104 - Parse Error: Identifier "Types" not declared in the current scope.
  res://addons/block_code/drag_manager/drag_manager.gd:104 - Parse Error: Identifier "Types" not declared in the current scope.
  res://addons/block_code/drag_manager/drag_manager.gd:114 - Parse Error: Could not find type "SnapPoint" in the current scope.
  res://addons/block_code/drag_manager/drag_manager.gd:123 - Parse Error: Could not find type "EntryBlock" in the current scope.
  modules/gdscript/gdscript.cpp:2726 - Failed to load script "res://addons/block_code/drag_manager/drag_manager.gd" with error "Parse error". (User)
  res://addons/block_code/block_code_plugin.gd:49 - Parse Error: Identifier "Types" not declared in the current scope.
  res://addons/block_code/block_code_plugin.gd:104 - Parse Error: Could not find type "BlockCode" in the current scope.
  res://addons/block_code/block_code_plugin.gd:104 - Parse Error: Could not find type "BlockCode" in the current scope.
  modules/gdscript/gdscript.cpp:2726 - Failed to load script "res://addons/block_code/block_code_plugin.gd" with error "Parse error". (User)
  File 'C:/Users/cranc/AppData/Local/Godot/editor_doc_cache.res' can't be loaded, as it uses a format version (6) or engine version (4.3) which are not supported by your engine version (4.2).
  Failed loading resource: C:/Users/cranc/AppData/Local/Godot/editor_doc_cache.res. Make sure resources have been imported by opening the project in the editor at least once.
  Missing required editor-specific import metadata for a texture (please reimport it using the 'Import' tab): 'res://.godot/imported/block_code_node.svg-ec44d62670f5745547380d82dd01a2b1.editor.meta'
  scene/resources/resource_format_text.cpp:448 - res://addons/block_code/ui/node_list/node_class_list/all_nodes.tres:5 - ext_resource, invalid UID: uid://dr8e0tvfxjy1f - using text path instead: res://icon.svg
  res://addons/block_code/block_code_plugin.gd:54 - Invalid set index 'undo_redo' (on base: 'MarginContainer (MainPanel)') with value of type 'EditorUndoRedoManager'.
  res://addons/block_code/block_code_plugin.gd:100 - Invalid call. Nonexistent function 'switch_scene' in base 'MarginContainer (MainPanel)'.
  res://addons/block_code/block_code_plugin.gd:105 - Invalid call. Nonexistent function 'switch_script' in base 'MarginContainer (MainPanel)'.
```

**Comment by wnbaum** [comment]
It seems that opening the editor for the first time with a plugin already enabled results in these errors, likely because the `.godot` directory hasn't been generated yet.

I'm assuming you cloned the repository and opened the project straight away which would have caused these errors. There is a small typo in the getting started instructions which may have led you astray. I fixed this in #104. It seems like this is mainly a Godot related error with no clear workaround - hopefully will be fixed in newer versions.

**Comment by manuq** [comment]
Editor lock fixed by https://github.com/endlessm/godot-block-coding/pull/115
